core/state/metadata.go

```diff
@@ -64,3 +64,42 @@ const (
 	// Note it doesn't account for a self-destruct which appoints itself as recipient.
 	BalanceDecreaseSelfdestructBurn BalanceChangeReason = 14
 )
+
+// TODO: to be removed, it's useful for logs while developing
+// String returns a string representation of the reason.
+func (r BalanceChangeReason) String() string {
```

Comment:
Feels like this would be a good addition to the core code directly, it's probably worth it.